### DISCUSSION TITLE
Download on Releases

### DISCUSSION THREAD

**Discussion by jamesdwilson** [ExistentialAudio/BlackHole#746] [Q&A] [answered]
I very much appreciate your hard work in creating the application! Thank you!

I realize that I have no expectation of anything at all, but it feels very anti-GPL to claim this is a GPL project, link over to your site, and donate-shame people when they try to download the build.

One should not literally have to click "I can't afford to donate" in order to download the release for free.

And then on top of that, you are requiring a first name, last name, email address, just to download.

I humbly request you just host the releases here on github and put a link to your site in addition to that if you want.

**Comment by devinroth** [comment]
On the contrary. I find it is very much in the spirit of the GPL license. And I disagree that I'm "shaming" people. I feel like the current solution is a good balance between receiving donations for all the hard work that I continue to put into this project while keeping it open source. There is too much emphases on "free" as in costs no money in "free and open source". In my opinion "free" is referring to the ability to change anything you want. Too many fantastic open source contributors that have made amazing contributions to technology are taken advantage of. It's a significant issue in the space. I have tried a number of different approaches and I find the balance works quite well right now. In fact, I get emails from people thanking me for keeping it free. I know I will not make everyone happy and that's okay. 

If you have any suggestions that will solve this issue in the open source community, I would love to know. 

I am always open to respectful conversation.

**Reply by jamesdwilson** [reply]
Hi Devin!

First and foremost, I want to again extend my sincere appreciation for the hard work and dedication you've invested in BlackHole.

The functionality it offers is robust, and the user experience is intuitive. It's evident that a lot of thought and effort has gone into its development.

I want to reiterate I am bringing up concern out of utmost respect for you and with a broader vision towards the betterment of the software community. 

The current setup for downloading BlackHole via your website, while well-intentioned, may pose some challenges that could potentially misalign with the core principles of Free and Open Source Software (FOSS) and the GNU General Public License (GPL). 
Here are my thoughts detailed below:

### Privacy Concerns
Privacy transcends a core value in the FOSS community; it's a fundamental human right acknowledged globally. The requirement to provide personal information to download the software unveils serious concerns, and I'd like to pose some challenges regarding this setup:

- **Legal Name vs. Identified Name:** What if a user provides a name they identify with, which may not correspond to their legal name? This is especially pertinent in the case of transgender individuals or those with non-binary gender identities, for whom legal names might not reflect their true identity. This scenario not only creates a complex situation concerning the accuracy and authenticity of the data collected but may also conflict with the inclusive and respectful ethos the FOSS community strives for.

- **Invalid Email Addresses:** If a user provides an incorrect or fake email address, how does this setup ensure the validity of the data collected? It seems like this requirement might lead to collecting inaccurate data, which not only questions the effectiveness of this approach but also potentially complicates compliance with privacy laws.

- **Invalid Name Submission:** Similarly, if someone provides an invalid name, how would this be handled? This further complicates the authenticity of the data collected and may pose additional challenges.

- **Data Storage and Compliance:** The collection of personal information necessitates secure storage and adherence to various privacy laws, including the General Data Protection Regulation (GDPR). How does the current setup ensure compliance with these laws, especially considering the legal and financial implications?

- **User Deterrence:** Some users might be deterred from downloading the software due to these privacy concerns, which contradicts the FOSS principle of easy and open access to software. Is this trade-off between potential financial support and user accessibility a considered aspect of the current setup?

- **Donation Shaming Concern:** The setup, where one has to click "I can't afford to donate" to access the download, may unintentionally evoke a sense of shame or discomfort. While it's crucial to seek community support, it might be more in the spirit of FOSS to do so in a way that doesn’t potentially alienate or discomfort individuals.

These challenges highlight intricate privacy concerns that extend beyond the immediate interaction on the download page. They encompass broader ethical, legal, and societal dimensions, which are crucial to consider for aligning with the FOSS philosophy and ensuring a user-friendly, respectful, and lawful interaction with the user community.

Your efforts towards creating and maintaining BlackHole are highly commendable. I am hopeful that this discussion can contribute positively towards the project’s continuous success while upholding the principles of FOSS and GPL.

Warm regards,

James

**Reply by devinroth** [reply, marked as answer]
First off, I have never shared any data with any 3rd parties nor will I ever. That is very clear in my privacy policy. 

There are plenty of FOSS projects that make money from selling advertising. I'm against that approach which is why I've been trying something different. 

The concern you have with user deterrence seems to be invalid based on my knowledge. 

I appreciate your concerns but I have to respectfully disagree with your proposed solution. 

Ultimately I have bills to pay, a family to feed. I would love nothing more than to work on open source projects and open source all my apps but I'm not independently wealthy. If some uber wealthy angel wants to support me for the rest of my life, I will commit to open sourcing all my apps (they can contact me at devinroth@existential.audio). But until then, I'm trying to find a balance between paying the bills and giving software away for free to those who need it. 

I see no unlawful, disrespectful, or unfriendly issues. I'm sorry you feel that way. 

Devin

**Reply by jamesdwilson** [reply]
Thank you for your perspective! I genuinely appreciate it.
